cli/commands.go

```diff
@@ -71,7 +71,7 @@ func Commands(metaPtr *meta.Meta) map[string]cli.CommandFactory {
 				CredentialBackends: map[string]logical.Factory{
 					"approle":  credAppRole.Factory,
 					"cert":     credCert.Factory,
-					"aws-ec2":  credAwsEc2.Factory,
```

Comment:
The current alias code in `vault/auth.go` will still allow scripts that do things like `vault auth-enable aws-ec2` if that's what you're concerned about (and mount them at `/auth/aws-ec2`, maintaining backwards compatibility), which I think is what you're suggesting.